src/components/Select/wrappers/SelectAsync/SelectAsync.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import flatMap from 'lodash/flatMap';
+import {useInfinityFetch} from '../../../utils/Select/useSelectInfinityFetch/useInfinityFetch';
+import {useSelectLoading} from '../../../utils/Select/useSelectLoading/useSelectLoading';
+import {SelectBasic} from '../../Select';
+import {SelectAsyncProps} from '../../types';
+
+export const SelectAsync = React.forwardRef<HTMLButtonElement, SelectAsyncProps>(
+    function SelectAsync<Option = any, Pagination = any>(
+        props: SelectAsyncProps<Option, Pagination>,
+        ref: React.Ref<HTMLButtonElement>,
+    ) {
+        const {options: fetcher} = props;
+
+        const {responses, onFetchInfinity, isLoadingInitial} = useInfinityFetch({
+            fetcher,
+        });
+
+        const mappedOptions = React.useMemo(
+            () => flatMap(responses, ({response}) => response),
+            [responses],
+        );
+
+        const asyncProps = useSelectLoading({
```

Comment:
what about using fetcher in case of using filter section?

Reply:
You should pass `onFilterChange` callback and recreate `fetcher` function while search pattern changes.
I can write ShowCase for this scenario.